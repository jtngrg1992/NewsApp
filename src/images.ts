const basePath = './assets/images';
export const Images = {
  calendar: require(`${basePath}/calendar.png`),
  next: require(`${basePath}/next.png`),
  menu: require(`${basePath}/menu.png`),
};
