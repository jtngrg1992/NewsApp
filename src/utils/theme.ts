export default {
  primaryColor: '#48e4c2',
  secondaryColor: '#9a9a9a',
  backgroundColor: '#fafafa',
  onPrimary: '#fff',
  onBackground: '#000',
  onBackgroundLight: '#808080',
  onSecondary: '#fff',
};
