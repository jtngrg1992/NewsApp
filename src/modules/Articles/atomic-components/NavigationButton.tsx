import React from 'react';
import {
  TouchableOpacity,
  Image,
  StyleSheet,
  ImageSourcePropType,
  TouchableOpacityProps,
} from 'react-native';
import theme from '@news/theme';

export type NavigationButtonProps = {
  image: ImageSourcePropType;
} & TouchableOpacityProps;
export const NavigationButton = ({
  image,
  style,
  ...otherProps
}: NavigationButtonProps) => {
  return (
    <TouchableOpacity
      style={[styles.container, StyleSheet.flatten(style)]}
      {...otherProps}>
      <Image source={image} resizeMode="contain" style={styles.icon} />
    </TouchableOpacity>
  );
};

const styles = StyleSheet.create({
  container: {
    paddingHorizontal: 10,
  },
  icon: {
    width: 20,
    tintColor: theme.onPrimary,
  },
});
