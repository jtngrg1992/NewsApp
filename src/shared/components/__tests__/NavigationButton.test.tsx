import {Images} from '@news/utils/images';
import {shallow, ShallowWrapper} from 'enzyme';
import React from 'react';
import {NavigationButton} from '../NavigationButton';

describe('unit tests for NavigationButton', () => {
  let wrapper: ShallowWrapper;

  const mockPress = jest.fn();
  beforeEach(() => {
    wrapper = shallow(
      <NavigationButton image={Images.calendar} onPress={mockPress} />,
    );
  });

  it('should render correctly', () => {
    expect(wrapper).toMatchSnapshot();
  });

  it('should render correct image in designated `Image` element', () => {
    const image = wrapper.find('Image');
    expect(image.prop('source')).toEqual({
      testUri: '../../../src/assets/images/calendar.png',
    });
  });
});
