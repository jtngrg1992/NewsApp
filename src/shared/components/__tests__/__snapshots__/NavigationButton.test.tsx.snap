// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`unit tests for NavigationButton should render correctly 1`] = `
<ForwardRef
  style={
    Array [
      Object {
        "paddingHorizontal": 10,
      },
      undefined,
    ]
  }
>
  <Image
    resizeMode="contain"
    source={
      Object {
        "testUri": "../../../src/assets/images/calendar.png",
      }
    }
    style={
      Object {
        "tintColor": "#fff",
        "width": 20,
      }
    }
  />
</ForwardRef>
`;
